inject-generator/src/main/java/io/avaje/inject/generator/ProcessingContext.java

```diff
@@ -205,7 +205,15 @@ static void processingOver(boolean over) {
   }
 
   static void writeSPIServicesFile() {
+    Optional.ofNullable(APContext.getProjectModuleElement())
+        .filter(m -> "io.avaje.inject.test".equals(m.getQualifiedName().toString()))
+        .ifPresent(m -> CTX.get().spiServices.remove(EVENTS_SPI));
+
     readExistingMetaInfServices();
+    if (CTX.get().spiServices.isEmpty()) {
```

Comment:
Are you happy with this change to only conditionally generate the services file @SentryMan ?

If so I think we can merge.

Reply:
Yeah I'm good with this